Gems/Atom/Feature/Common/Code/Include/Atom/Feature/Mesh/MeshInfo.h

```diff
@@ -0,0 +1,141 @@
+/*/
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <Atom/RHI.Reflect/Handle.h>
+#include <Atom/RHI.Reflect/IndexFormat.h>
+#include <Atom/RHI.Reflect/ShaderSemantic.h>
+#include <Atom/RHI.Reflect/VertexFormat.h>
+#include <Atom/RHI/Buffer.h>
+#include <Atom/RHI/BufferView.h>
+#include <Atom/RHI/IndexBufferView.h>
+#include <Atom/RHI/StreamBufferView.h>
+#include <Atom/RPI.Public/Buffer/BufferSystemInterface.h>
+#include <Atom/RPI.Public/Model/UvStreamTangentBitmask.h>
+#include <Atom/RPI.Reflect/Shader/ShaderOptionGroup.h>
+
+namespace AZ::Render
+{
+    using MeshInfoHandle = RHI::Handle<int32_t>;
+    using MeshInfoHandleList = AZStd::vector<MeshInfoHandle>;
+
+    // Utility to access the Geometry - Data from a mesh without the input-assembly
+    struct BufferViewIndexAndOffset
+    {
+        // Streambuffer-view and format
+        AZ::RHI::StreamBufferView m_streamBufferView{};
+        RHI::VertexFormat m_vertexFormat = RHI::VertexFormat::Unknown;
+
+        // Buffer-View, offset and bindless Read index for the same data, needed to access the data with the MeshInfo-indices
+        RHI::Ptr<AZ::RHI::BufferView> m_bufferView = nullptr;
+        uint32_t m_byteOffset = 0;
+        AZStd::unordered_map<int, uint32_t> m_bindlessReadIndex = {};
+
+        // utility function to create an entry from a generic RHI buffer
+        static BufferViewIndexAndOffset Create(
+            RHI::Buffer* rhiBuffer, const uint32_t byteOffset, RHI::VertexFormat vertexFormat, bool createStreamBufferView)
```

Comment:
Could the parameter `rhiBuffer` be made const? All method which are called in `CreateInternal` (`GetDescriptor`, `GetBufferView`) are also const.

Reply:
indeed it can!